### PR TITLE
Bloody soles for digitigrade legs

### PR DESCRIPTION
**Pull request by Seris02**
## About The Pull Request

Adds in bloody shoe overlays for digitigrade legs. Also edits the digi bloody sole image to look slightly more natural.

## Why It's Good For The Game

Because it's really annoying to look at the plantigrade ones over the digitigrade legs, especially when walking in almost invisible bloody footprints.

## Changelog
:cl:
add: Added digitigrade bloody shoes overlays
imageadd: added the icons for digitigrade bloody soles.
/:cl:

### CODE REVIEW

File: code/datums/components/bloodysoles.dm [hunk 1]
```diff
@@ -275,6 +277,14 @@
...
 
 /datum/component/bloodysoles/feet/on_step_blood(datum/source, obj/effect/decal/cleanable/pool)
+	//SKYRAT EDIT ADDITION BEGIN - DIGI_BLOODSOLE
+	if(!bloody_feet)
+		var/mob/living/carbon/H = parent
+		if (DIGITIGRADE in H.dna.species.species_traits)
+			bloody_feet = mutable_appearance('modular_skyrat/modules/digi_bloodsole/icons/effects/blood.dmi', "shoeblood", SHOES_LAYER)
+		else
+			bloody_feet = mutable_appearance('icons/effects/blood.dmi', "shoeblood", SHOES_LAYER)
+	//SKYRAT EDIT ADDITION END
```
Comment (atakiya) [inline code/datums/components/bloodysoles.dm:288]: Why was this moved from `Initialize()`?

Comment (Seris02) [reply code/datums/components/bloodysoles.dm:288]: It gets called too soon in both initialize() and update_icon(), because it gets called before copy_to() where the digitigrade prefs are put onto the character, so putting it in on_step_blood() is the only option.

**Comment by Azarak** [comment]
Make so worn overlays gets a mutant style argument passed and use it instead of checking species

Comment in the code on why the bloody soles overlays is initialized on step

**Review by Azarak — APPROVED** [review]
